python/paddle/nn/clip.py

```diff
@@ -540,6 +540,18 @@ def _dygraph_clip(self, params_grads):
             params_and_grads.append((p, new_grad))
         return params_and_grads
 
+    def _pir_clip(self, params_grads):
+        params_and_grads = []
+        for p, g in params_grads:
+            if g is None:
+                continue
+            if getattr(p, 'stop_gradient', True) is True:
```

Comment:
```suggestion
            if getattr(p, 'need_clip', True) is False:
```
这行要保持和静态图的行为一致，都是去检查 `need_clip` 属性